viziphant/unitary_event_analysis.py

```diff
@@ -180,203 +150,227 @@ def plot_ue(data, Js_dict, sig_level, binsize, winsize, winstep,
+    xlim_ax4 = axes[4].get_xlim()[1]
+    alpha_pos_text = axes[4].text(xlim_ax4, Js_sig, r'$\alpha +$', color='r',
+                                  horizontalalignment='right',
+                                  verticalalignment='bottom')
+    alpha_neg_text = axes[4].text(xlim_ax4, -Js_sig, r'$\alpha -$', color='g',
+                                  horizontalalignment='right',
+                                  verticalalignment='top')
+    axes[4].set_yticks([ue.jointJ(1 - significance_level), ue.jointJ(0.5),
+                        ue.jointJ(significance_level)])
+    # Try '1 - 0.34' to see the floating point errors
+    axes[4].set_yticklabels(np.round([1 - significance_level, 0.5,
+                                      significance_level], decimals=6))
+
+    # autoscale fix to mind the text positions.
+    # See https://stackoverflow.com/questions/11545062/
+    # matplotlib-autoscale-axes-to-include-annotations
+    plt.get_current_fig_manager().canvas.draw()
+    for text_handle in (alpha_pos_text, alpha_neg_text):
+        bbox = text_handle.get_window_extent()
+        bbox_data = bbox.transformed(axes[4].transData.inverted())
+        axes[4].update_datalim(bbox_data.corners(), updatex=False)
+    axes[4].autoscale_view()
+
+    mask_nonnan = ~np.isnan(Js_dict['Js'])
+    significant_win_idx = np.nonzero(Js_dict['Js'][mask_nonnan] >= Js_sig)[0]
+    t_winpos_significant = t_winpos[mask_nonnan][significant_win_idx]
+    axes[5].set_title('Unitary Events')
+    if len(t_winpos_significant) > 0:
+        for n in range(n_neurons):
+            if not neurons_participated[n]:
+                continue
+            for tr, data_tr in enumerate(spiketrains):
+                indices = np.unique(Js_dict['indices'][f'trial{tr}'])
+                indices_significant = []
+                for t_sig in t_winpos_significant:
+                    mask = (indices * bin_size >= t_sig
+                            ) & (indices * bin_size < t_sig + win_size)
+                    indices_significant.append(indices[mask])
+                indices_significant = np.hstack(indices_significant)
+                indices_significant = np.unique(indices_significant)
+                # does nothing if indices_significant is empty
+                axes[5].plot(indices_significant * bin_size,
+                             np.full_like(indices_significant,
+                                          fill_value=n * n_trials + tr),
+                             ms=plot_params['ms'], marker='s', ls='',
+                             mfc='none', mec='r')
+    axes[5].set_xlabel(f'Time ({t_winpos.dimensionality})',
+                       fontsize=plot_params['fsize'])
+    for key in plot_params['events'].keys():
+        for event_time in plot_params['events'][key]:
+            axes[5].text(event_time - 10 * pq.ms,
+                         axes[5].get_ylim()[0] - 35, key,
+                         fontsize=plot_params['fsize'], color='r')
```

Comment:
The labels are not always visible in the figure. Perhaps one should consider adjusting the placement?

Reply:
I don't know how to fix this issue efficiently. Ideally, `viziphant.event.add_event` should be used. But currently, it's not possible.
Perhaps, this can be a separate PR once either of us comes up with an appropriate solution.